features/collection/api/src/main/java/org/opennms/netmgt/collection/support/builder/Resource.java

```diff
@@ -99,4 +101,13 @@ public interface Resource {
      */
     public Resource resolve();
 
+
+    default Map<String, String> getTags() {
+        return new HashMap<>();
```

Comment:
Same here. Use `Collections.emptyMap()`. This will make a call to the add methods fail as expected.